README.md

```diff
@@ -45,8 +45,13 @@ returns the length of the shortest tour (not the tour itself).
 
 Test your new function; I've provided some basic testing code in `code.test.js`.
 
+ I asked question in office hours, impromptu hours. I referenced https://www.w3docs.com/snippets/javascript/how-to-randomize-shuffle-a-javascript-array.html which I learned and used the Fisher Yates algorithm, I also looked at this website for randomization as well, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random. I also referenced my old TSP Held Karp and pancake sort and dynamic euler.
+I certify that I have listed all sources used to complete this exercise, including the use of any Large Language Models. All of the work is my own, except where stated otherwise. I am aware that plagiarism carries severe penalties and that if plagiarism is suspected, charges may be filed against me without prior notice.
+
 ## Runtime Analysis
 
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+For the worst case time complexity, overall, for most of these loops they have a runtime of O(v) or they have a runtime of a maximum of O(v^2), however, one major factor is conquering these times. This would be the stopping criteria, which stops after v! iterations. This leads to the actual worst case time complexity being O(v!). For memory complexity, although we use some temporary arrays which are O(v), we will only have a space complexity of about O(v^2); this O(v^2) comes from the distance_matrix.
```

Comment:
How much work do you have to do for each iteration? And doesn't that result in a worse time complexity than Held-Karp?